books/views.py

```diff
@@ -60,8 +60,28 @@ def get(self, request):
 
 
 class BookDetailView(TemplateView):
+    form_class = RequestBookForm
     template_name = "books/book_detail.html"
 
     def get(self, request, pk):
         book = Book.objects.get(pk=pk)
-        return render(request, self.template_name, {"book": book})
+        form = self.form_class()
+        return render(
+            request, self.template_name, {"book": book, "form": form}
+        )
+
+    def post(self, request, pk):
+        book = Book.objects.get(pk=pk)
+        form = self.form_class(request.POST)
+        if form.is_valid():
+            if request.user.is_authenticated:
```

Comment:
You shouldn't have to check this in your views most of the time. Django has a decorator you can use to protect specific endpoints and make them for authenticated users only. Have a look at:

https://docs.djangoproject.com/en/2.1/topics/auth/default/#the-login-required-decorator

You can use it even with classes:
https://docs.djangoproject.com/en/2.1/topics/class-based-views/intro/#decorating-the-class